### PR TITLE
Recovery Services Backup Initial PR

### PR DESCRIPTION
**Pull request by dragonfly91**
Hi,

This PR comprises of a draft version of the new module - RecoveryServices.Backup. We wanted to get an early feedback on various aspects of the code / design etc. Please review the PR but do NOT merge the PR - since this changeset contains only a few cmdlets. Look forward to your feedback.

Thanks,
Anudeep.

### CODE REVIEW

**Comment by hovsepm** [comment]
@dragonfly91 What is relationship between changes in AzureBackup and RecoveryServices and adding RecoveryServices.Backup module to the repository? could they go in independent of each other?

**Comment by hovsepm** [comment]
@dragonfly91 please change the overall folder structure. Take a look to src\ResourceManager\RecoveryServices project -> product and Test should be in a different folders.

**Comment by hovsepm** [comment]
Your cmdlets should not define 2 output types for lists and for single types.
```
OutputType(typeof(List<AzureRmRecoveryServicesContainerBase>), typeof(AzureRmRecoveryServicesContainerBase))
```
do not define List, instead in when you do write object specify that it can be an enumerable collection. e.g. 
```
WriteObject(containerModels, enumerateCollection: true);
```

**Comment by hovsepm** [comment]
@dragonfly91 you should NOT use things like 
```
Thread.Sleep(15 * 1000); 
```
things. Use TestMockSupport.Delay method from Commands.Common project.

**Comment by hovsepm** [comment]
Move Commands.RecoveryServices.Backup.Helpers to a separate folder under RecoveryServices.Backup folder. Do not put projects in nested subfolders.

**Comment by hovsepm** [comment]
Why do you have so many library packages that are only referenced in one module?

**Comment by hovsepm** [comment]
you should not reference anything from GAC (e.g. Commands.RecoveryServices.Backup.Cmdlets.csproj)
```
    <Reference Include="System.Management.Automation, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
      <SpecificVersion>False</SpecificVersion>
      <HintPath>C:\Windows\Microsoft.NET\assembly\GAC_MSIL\System.Management.Automation\v4.0_3.0.0.0__31bf3856ad364e35\System.Management.Automation.dll</HintPath>
    </Reference>
```

remove the hint path.

**Comment by hovsepm** [comment]
@dragonfly91 since you guys added your module to AzureRM and also added psd1 file for RecoveryServices.Backup you will need to regenerate wxi file and include it to this PR as well.

**Comment by hovsepm** [comment]
on-demand job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/690/ 
Sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/380/

**Comment by hovsepm** [comment]
on-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/716/

**Comment by hovsepm** [comment]
sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/385/

**Comment by hovsepm** [comment]
new sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/388/

**Comment by hovsepm** [comment]
another one for sing job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/389/

**Comment by hovsepm** [comment]
sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/391/

**Comment by hovsepm** [comment]
on-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/720/

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Commands.RecoveryServices.Backup.Helpers.csproj [hunk 1]
```diff
@@ -24,10 +24,14 @@
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
     <DebugType>pdbonly</DebugType>
     <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
+    <OutputPath>..\..\..\Package\Release\ResourceManager\AzureResourceManager\AzureRM.RecoveryServices.Backup\</OutputPath>
```
Comment (hovsepm) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Commands.RecoveryServices.Backup.Helpers.csproj]: no need for this

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Logger/Commands.RecoveryServices.Backup.Logger.csproj [hunk 2]
```diff
@@ -24,10 +24,14 @@
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
     <DebugType>pdbonly</DebugType>
     <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
+    <OutputPath>..\..\..\Package\Release\ResourceManager\AzureResourceManager\AzureRM.RecoveryServices.Backup\</OutputPath>
```
Comment (hovsepm) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Logger/Commands.RecoveryServices.Backup.Logger.csproj]: remove this line

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Models/Commands.RecoveryServices.Backup.Models.csproj [hunk 3]
```diff
@@ -24,10 +24,14 @@
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
     <DebugType>pdbonly</DebugType>
     <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
+    <OutputPath>..\..\..\Package\Release\ResourceManager\AzureResourceManager\AzureRM.RecoveryServices.Backup\</OutputPath>
```
Comment (hovsepm) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Models/Commands.RecoveryServices.Backup.Models.csproj]: revert this line

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Commands.RecoveryServices.Backup.Providers.csproj [hunk 4]
```diff
@@ -26,10 +26,14 @@
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
     <DebugType>pdbonly</DebugType>
     <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
+    <OutputPath>..\..\..\Package\Release\ResourceManager\AzureResourceManager\AzureRM.RecoveryServices.Backup\</OutputPath>
```
Comment (hovsepm) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Commands.RecoveryServices.Backup.Providers.csproj]: revert this line

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.ServiceClientAdapter/Commands.RecoveryServices.Backup.ServiceClientAdapter.csproj [hunk 5]
```diff
@@ -24,10 +24,14 @@
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
     <DebugType>pdbonly</DebugType>
     <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
+    <OutputPath>..\..\..\Package\Release\ResourceManager\AzureResourceManager\AzureRM.RecoveryServices.Backup\</OutputPath>
```
Comment (hovsepm) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.ServiceClientAdapter/Commands.RecoveryServices.Backup.ServiceClientAdapter.csproj]: revert this line

**Comment by hovsepm** [comment]
sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/393/

**Comment by markcowl** [comment]
signing: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/398/

**Pull request merged**